tsdb/engine/tsm1/compact_test.go

```diff
@@ -2731,49 +2731,50 @@ func TestDefaultPlanner_PlanOptimize_Test(t *testing.T) {
 		},
 	}
 
-	expectedFullyCompacted := func(cp *tsm1.DefaultPlanner, reasonExp string, testName string) {
+	expectedFullyCompacted := func(cp *tsm1.DefaultPlanner, reasonExp string) {
 		compacted, reason := cp.FullyCompacted()
-		require.Equal(t, reason, reasonExp, "fullyCompacted reason", testName)
-		require.True(t, compacted, "is fully compacted", testName)
+		require.Equal(t, reason, reasonExp, "fullyCompacted reason")
+		require.True(t, compacted, "is fully compacted")
 
 		_, cgLen := cp.PlanLevel(1)
-		require.Zero(t, cgLen, "compaction group length; PlanLevel(1)", testName)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(1)")
 		_, cgLen = cp.PlanLevel(2)
-		require.Zero(t, cgLen, "compaction group length; PlanLevel(2)", testName)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(2)")
 		_, cgLen = cp.PlanLevel(3)
-		require.Zero(t, cgLen, "compaction group length; PlanLevel(3)", testName)
+		require.Zero(t, cgLen, "compaction group length; PlanLevel(3)")
 
 		tsmP, pLenP := cp.Plan(time.Now().Add(-time.Second))
-		require.Zero(t, len(tsmP), "compaction group; Plan()", testName)
-		require.Zero(t, pLenP, "compaction group length; Plan()", testName)
+		require.Zero(t, len(tsmP), "compaction group; Plan()")
+		require.Zero(t, pLenP, "compaction group length; Plan()")
 
 		cgroup, cgLen, genLen := cp.PlanOptimize()
-		require.Equal(t, []tsm1.CompactionGroup(nil), cgroup, "compaction group", testName)
-		require.Zero(t, cgLen, "compaction group length", testName)
-		require.Zero(t, genLen, "generation count", testName)
+		require.Equal(t, []tsm1.CompactionGroup(nil), cgroup, "compaction group")
+		require.Zero(t, cgLen, "compaction group length")
+		require.Zero(t, genLen, "generation count")
 	}
 
 	for _, test := range areFullyCompactedTests {
-		ffs := &fakeFileStore{
-			PathsFn: func() []tsm1.FileStat {
-				return test.fs
-			},
-		}
+		t.Run(test.name, func(t *testing.T) {
+			ffs := &fakeFileStore{
+				PathsFn: func() []tsm1.FileStat {
+					return test.fs
+				},
+			}
 
-		if len(test.bc) > 0 {
-			err := ffs.SetBlockCounts(test.bc)
-			require.NoError(t, err, "setting block counts")
-		}
+			if len(test.bc) > 0 {
+				err := ffs.SetBlockCounts(test.bc)
+				require.NoError(t, err, "setting block counts")
+			}
 
-		cp := tsm1.NewDefaultPlanner(ffs, tsdb.DefaultCompactFullWriteColdDuration)
-		expectedFullyCompacted(cp, test.expectedFullyCompactedReasonExp, test.name)
+			cp := tsm1.NewDefaultPlanner(ffs, tsdb.DefaultCompactFullWriteColdDuration)
+			expectedFullyCompacted(cp, test.expectedFullyCompactedReasonExp)
 
-		// Reverse test files and re-run tests
-		slices.Reverse(test.fs)
-		cp = tsm1.NewDefaultPlanner(ffs, tsdb.DefaultCompactFullWriteColdDuration)
-		expectedFullyCompacted(cp, test.expectedFullyCompactedReasonExp, test.name)
+			// Reverse test files and re-run tests
+			slices.Reverse(test.fs)
```

Comment:
Do we also need to reverse the block counts here?

Reply:
@davidby-influx ah yes I believe so